superset-frontend/src/SqlLab/components/TabbedSqlEditors/TabbedSqlEditors.test.jsx

```diff
@@ -163,23 +169,32 @@ describe('TabbedSqlEditors', () => {
       wrapper.instance().props.actions.removeQueryEditor.getCall(0).args[0],
     ).toBe(queryEditors[0]);
   });
-  it('should add new query editor', () => {
-    wrapper = getWrapper();
-    sinon.stub(wrapper.instance().props.actions, 'addQueryEditor');
-
-    wrapper.instance().newQueryEditor();
-    expect(
-      wrapper.instance().props.actions.addQueryEditor.getCall(0).args[0].name,
-    ).toContain('Untitled Query');
+  it('should add new query editor', async () => {
+    const { getAllByLabelText } = setup(mockedProps);
+    fireEvent.click(getAllByLabelText('Add tab')[0]);
+    const actions = store.getActions();
+    await waitFor(() =>
+      expect(actions).toContainEqual({
+        type: 'ADD_QUERY_EDITOR',
+        queryEditor: expect.objectContaining({
+          name: expect.stringMatching(/Untitled Query (\d+)+/),
+        }),
+      }),
+    );
   });
-  it('should properly increment query tab name', () => {
-    wrapper = getWrapper();
-    sinon.stub(wrapper.instance().props.actions, 'addQueryEditor');
-    const newTitle = newQueryTabName(wrapper.instance().props.queryEditors);
-    wrapper.instance().newQueryEditor();
-    expect(
-      wrapper.instance().props.actions.addQueryEditor.getCall(0).args[0].name,
-    ).toContain(newTitle);
+  it('should properly increment query tab name', async () => {
```

Comment:
RTL best practices recommend testing what the user interacts with instead of implementation details. Here's a quote from their site:

> You want to write maintainable tests that give you high confidence that your components are working for your users. As a part of this goal, you want your tests to avoid including implementation details so refactors of your components (changes to implementation but not functionality) don’t break your tests and slow you and your team down.

In this case, it would be better if we test what appears on the screen like checking if there's a new tab with the right title. Even if we change the action name or other logic in the future, the expected result would be the same. This is not a blocker for me though, it's just a suggestion.

Reply:
Good suggestion. However, it's a little complicated to setup entire SqlLab redux state to simulate the behavior in the screen. It'd be better to minimize the redux usage in the future for better screen testing environment.